pkg/plugins/bootstrap/k8s/plugin.go

```diff
@@ -176,198 +160,26 @@ func (p *plugin) Order() int {
-// Controllers must be explicitly added by calling AddControllers.
-func (kmw *kubeManagerWrapper) Add(runnable kube_manager.Runnable) error {
-	if ctrler, ok := runnable.(kube_ctrlr.Controller); ok {
-		kmw.controllers = append(kmw.controllers, ctrler)
-		return nil
-	}
-	return kmw.Manager.Add(runnable)
-}
-
-// AddControllers calls Manager.Add on any accumulated controllers.
-func (kmw *kubeManagerWrapper) AddControllersToManager() error {
-	for _, c := range kmw.controllers {
-		if err := kmw.Manager.Add(c); err != nil {
-			return errors.Wrap(err, "add controller error")
-		}
-	}
-	return nil
-}
-
 type kubeComponentManager struct {
-	*kubeManagerWrapper
-	oldLeaderElectionNamespace string
-	leaderComponents           []component.Component
-	gracefulComponents         []component.GracefulComponent
+	kube_ctrl.Manager
+	gracefulComponents []component.GracefulComponent
 }
 
 var _ component.Manager = &kubeComponentManager{}
 
-type leaderAnnotation struct {
-	HolderIdentity       string `json:"holderIdentity"`
-	LeaseDurationSeconds int    `json:"leaseDurationSeconds"`
-	AcquireTime          string `json:"acquireTime"`
-	RenewTime            string `json:"renewTime"`
-	LeaderTransitions    int    `json:"leaderTransistions"`
-}
-
-var blockerHolderId = "cp-leader-lock-transition"
-var oldLeaderConfigMapName = "kuma-cp-leader"
-
-func makeOldLockAnnotation() string {
-	nowStr := time.Now().Format(time.RFC3339)
-	annot := &leaderAnnotation{
-		HolderIdentity:       blockerHolderId,
-		LeaseDurationSeconds: 99999999999999999,
-		AcquireTime:          nowStr,
-		RenewTime:            nowStr,
-		LeaderTransitions:    0,
-	}
-
-	annotJson, _ := json.Marshal(annot)
-	return string(annotJson)
-}
-
-// startLeaderComponents adds all components that need leader election to the
-// Manager. The Manager should be running already, any components added after
-// that are started immediately.
-func (cm *kubeComponentManager) startLeaderComponents() error {
-	for _, c := range cm.leaderComponents {
-		if err := cm.Manager.Add(&componentRunnableAdaptor{Component: c}); err != nil {
-			return errors.Wrap(err, "add component error")
-		}
-	}
-	return cm.kubeManagerWrapper.AddControllersToManager()
-}
-
-// Previous versions of kuma-cp used a timeout lock for leader election. We now
-// keep the election for the lifetime of the pod. This function forces any previous
-// style leader to see itself as having lost its election, and locks out any
-// further old leaders.
-//
-// Only call this after acquiring new-style leader election, so as to only contend
-// with old leaders over old locks.
-func (cm *kubeComponentManager) forceTakeOldLock(ctx context.Context) error {
-	log.Info("checking for deprecated leader locks")
-	client := cm.Manager.GetClient()
-	ns := cm.oldLeaderElectionNamespace
-
-	pod := &kube_core.Pod{}
-	if err := client.Get(ctx, kube_client.ObjectKey{
-		Namespace: ns,
-		Name:      os.Getenv("POD_NAME"),
```

Comment:
Probably we don't need `POD_NAME` for CP deployment?

Reply:
removed